src/language-server/config/config.ts

```diff
@@ -79,7 +81,29 @@ const clientConfig = z.object({
 export type ClientConfigFormat = z.infer<typeof clientConfig>;
 
 const roverConfig = z.object({
-  bin: z.string().optional(),
+  bin: z
+    .preprocess(
+      (val) => val || which.sync("rover", { nothrow: true }) || undefined,
+      z.string({
+        message:
+          "Rover binary not found. Please either install it system-wide in PATH, or provide the `bin` option. Please also ensure that the binary is executable.",
```

Comment:
```suggestion
          "Rover binary not found. Please install it system-wide in PATH or provide the `bin` option and ensure the binary is executable.",
```

Feel free to tweak this further, but I'd recommend combining both sentences to remove the double "please" call-to-action here.

Reply:
The "executable" part applies to both "or paths" of the sentence before, so it's `(in PATH or bin option) AND executable. It seemed less ambiguous to me to keep them separated - but I can remove the second `please`.
```suggestion
          "Rover binary not found. Please either install it system-wide in PATH, or provide the `bin` option. Also ensure that the binary is executable.",
```